### PR TITLE
GroupBy Rank SegFault Fix - astype instead of view

### PR DESCRIPTION
**Pull request by WillAyd**
xref #19679 and #19481

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Can you merge in master? The timeouts will hopefully be fixed...

**Comment by WillAyd** [comment]
adding @jreback for merge

File: pandas/_libs/groupby_helper.pxi.in [hunk 1]
```diff
@@ -531,7 +531,7 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
     # each label corresponds to a different group value,
     # the mask helps you differentiate missing values before
     # performing sort on the actual values
-    _as = np.lexsort(order).view(dtype=np.int64)
+    _as = np.lexsort(order).astype(np.int64)
```
Comment (chris-b1) [inline pandas/_libs/groupby_helper.pxi.in]: add `copy=False` which will elide the copy on a 64 bit platform.

**Comment by jreback** [comment]
sure can merge on green.

**Comment by jreback** [comment]
thanks again @WillAyd

**Pull request merged**